treesitter/default.nix

```diff
@@ -0,0 +1,79 @@
+flake-utils: {
+  yarn2nix,
+  yarn2nix-moretea,
+  mkYarnPackage,
+  nodePackages,
+  python3,
+  stdenv,
+  bundlerEnv,
+  ruby,
+  lib,
+  coreutils,
+  fetchzip,
+  xcbuild,
+  nodejs-16_x,
+  ...
+} @pkgs:
+
+  let
+    # treesitter doesn't work with node 17/18
+    treesitterNode = nodejs-16_x;
+
+    # converts /nix/store/.../foo.tabry to "foo"
+    commandNameFromTabryFilename = fileName: 
+      (builtins.replaceStrings [".tabry"] [""] (builtins.baseNameOf fileName));
+
+    formatJsonFilename = fileName: 
+      (commandNameFromTabryFilename fileName) + ".json";
+
+    tabryCompilerJsPath = "/libexec/tree-sitter-tabry/deps/tree-sitter-tabry/tabry-compile.js";
+
+    # This is a function that takes a .tabry file
+    # and returns a derivation that compiles that
+    # .tabry file into the tabry .json file
+    compileTabryFile = inFile: stdenv.mkDerivation {
```

Comment:
This isn't used anywhere else in the commit; what's the use case for it?

Reply:
It's used in https://github.com/evanbattaglia/tabry/pull/3 to compile `.tabry` files from people's home-manager config,

for example:
```
config.programs.tabry = {
  enable = true;
  tabryFiles = [
    ./rapture.tabry
    ./aws.tabry
  ]
}
```

this function is useful for compiling `rapture.tabry` and `aws.tabry`